### PR TITLE
Pathing issue

### LINKED ISSUE #20328: Path.intersects_path sometimes returns incorrect values
### Bug report

**Bug summary**

The Path.intersects_path function sometimes returns True when it should return False.  I've noticed this only when some vertices in the path deviate slightly from round numbers (as sometimes arises from floating point imprecision).

**Code for reproduction**

```python
from matplotlib.path import Path

path_0 = Path([
    [1, -1],
    [0, -1],
    [0, -2],
    [1, -2],
])
path_1 = Path([
    [1, 2],
    [0, 2],
    [0, 1],
    [0.999999999999, 1],
])

print(Path.intersects_path(path_0, path_1))
```

**Actual outcome**

```
True
```

**Expected outcome**

```
False 
```

Note that the two paths are completely separated by a large margin along the y axis, so do not intersect.

Additional observations:
* Choosing different y values for the paths or translating the entire configuration by choosing different x values does not change the result.
* Reducing the precision of the last vertex of `path_1` (e.g. making it 1 or 0.9999) changes the result to the expected `False`.

**Matplotlib version**

  * Operating system:  OS X 10.15.1
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`):  3.4.2 (bug exists with 3.1.0 as well)
  * Matplotlib backend (`print(matplotlib.get_backend())`):  MacOSX
  * Python version:  3.7.10

Matplotlib installed using pip in a conda virtual environment (`conda activate mpl_bug \ pip install matplotlib`) with no other dependencies installed.

### PR DESCRIPTION
**Pull request by diegopetrola**
## PR Checklist

Closes #20328

What I did:

- Implemented the code to solve the issue
- Added tests for the cases mentioned in #20328

I will wait for the automated checks to pass before asking for a review.
Edit: ready for review.

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

File: lib/matplotlib/tests/test_path.py [hunk 1]
```diff
@@ -410,6 +410,16 @@ def test_path_intersect_path(phi):
     b = transform.transform_path(Path([(0, 1), (0, 2), (0, 5)]))
     assert a.intersects_path(b) and b.intersects_path(a)
 
+    # test added to adress issue #20328
```
Comment (jklymak) [inline lib/matplotlib/tests/test_path.py:413]: Can you say what #20328 was?  This code may outlive GitHub and this comment won't be helpful...

**Review by jklymak — APPROVED** [review]
Aside from the nit about the comment...

File: lib/matplotlib/tests/test_path.py (hunk 1 above)
Comment (diegopetrola) [reply lib/matplotlib/tests/test_path.py]: Good suggestion, I rewrote the comments to be more alike the ones already present in the tests.

**Comment by jklymak** [comment]
Thanks @diegopetrola!

**Comment by QuLogic** [comment]
Did you not want this backported?

**Comment by jklymak** [comment]
(me?  I'm fine if we back port, but didn't think it was a regression...)

**Comment by QuLogic** [comment]
Yes, you milestoned the closed issue for 3.4.3.

**Comment by jklymak** [comment]
@meeseeksdev backport to 3.4.3

**Comment by jklymak** [comment]
@meeseeksdev backport to 3.4.x

**Comment by jklymak** [comment]
@meeseeksdev backport to v3.4.x

**Pull request merged**